### PR TITLE
[5.4] Add Model::refresh()

### PR DESCRIPTION
**Pull request by themsaid**
This method eliminates the need for:

```php
$user = $user->fresh()
```

You can do this instead:

```php
$user->refresh()
```

### CODE REVIEW

**Comment by mattstauffer** [comment]
Woo!

**Comment by calebporzio** [comment]
@themsaid  The actual usage would be: 
`$user->refresh()` ya?

**Comment by calebporzio** [comment]
Wondering if there is a way to not have to call `->load(...` every time I want to `refresh()`. Like `->fresh()` has optional "with" param... possibly same deal here?

**Comment by themsaid** [comment]
refresh() will reload all relationships currently loaded automatically.

**Comment by calebporzio** [comment]
got it, cool!

**Comment by morloderex** [comment]
@themsaid what is the different between fresh and refresh here? 

couldn't refresh just call fresh?

**Comment by decadence** [comment]
@morloderex fresh reload relations based on $with parameter, refresh reload all relations based on current model loaded relations.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -944,6 +944,23 @@ public function fresh($with = [])
     }
 
     /**
+     * Reload the current model instance with fresh attributes from the database.
+     *
+     * @param  array|string  $with
```
Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Model.php]: No param

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -944,6 +944,22 @@ public function fresh($with = [])
...
+     */
+    public function refresh()
+    {
+        if (! $this->exists) {
+            return;
+        }
+
+        $this->load(array_keys($this->relations));
+
+        $this->setRawAttributes(static::find($this->getKey())->attributes);
```
Comment (JosephSilber) [inline src/Illuminate/Database/Eloquent/Model.php]: Should this be `findOrFail`? What should happen if the model has been deleted in the meantime?

Comment (themsaid) [inline src/Illuminate/Database/Eloquent/Model.php]: If the model was deleted exists would return false and the method will just bail.

**Comment by blackness90** [comment]
Where are unit tests?

**Comment by JosephSilber** [comment]
I meant if it was deleted somewhere else, not in this request.

**Comment by themsaid** [comment]
@JosephSilber good point, switched to using `findOrFail`

**Comment by themsaid** [comment]
@rorc you can use `fresh()` for that.

**Comment by rorc** [comment]
@themsaid But, if i want reload only current relations?

**Comment by themsaid** [comment]
@rorc just call `refresh()` and return the original value of yours, thats' the point :)

**Comment by rorc** [comment]
@themsaid Sigh, except 1 line i have to use 2 lines? Are you kidding? 😒
```php
$selection->refresh();

return view('admin.selections.list-item')->with('selection', $selection);
```

**Comment by themsaid** [comment]
@rorc haha .. well I think I have an idea

**Comment by rorc** [comment]
@themsaid Why not just return the куакуырув model? 😉

**Comment by Vasiliy-Bondarenko** [comment]
plus 1 vote for 
return $this;
instead of void.

Very useful for chaining in tests.
```php
        $comment->refresh(); // redundant line here!
        $this->assertCount(1, $comment->replies);
```

**Pull request merged**